includes/class-wcs-att-cart.php

```diff
@@ -20,7 +20,29 @@
  */
 class WCS_ATT_Cart {
 
+	/**
+	 * Flag to ensure hooks can be added only once.
+	 * @var bool
+	 */
+	private static $added_hooks = false;
```

Comment:
This system should be unnecessary. WordPress will only ever add the same callback with the same priority once, see `_wp_filter_build_unique_id()` and `WP_Hook::add_filter()`.